src/users/user-manager.js

```diff
@@ -3,49 +3,84 @@ var userModels = require('./user-models.js');
-        outreach: 0,
-        professor: 0,
-        staff: 0,
-        misc: 0
-      },
-      role: 'Student',
-      studentId: reqData.studentId
-    });
-
-  } else if (reqData.role == 'Admin') {
-    // Create new admin.
-    user = new Admin({
-      email: reqData.email,
-      firstName: reqData.firstName,
-      lastName: reqData.lastName,
-      role: 'Admin'
-    });
+/**
+ * Callback for sending the response to the client.
+ *
+ * @function createResponse
+ * @param {Object} err - The error.
+ * @param {Number} id - The student ID.
+ */
 
-  } else {
-  // TODO(ryanfaulkenberry100): Handle error case where role is incorrect.
+/**
+ * Given User information, create a User.
+ * @param {Object} data - The request body data.
+ * @param {String} data.role - The user role.
+ * @param {String} data.email - The user email address.
+ * @param {String} data.password - The user password.
+ * @param {createResponse} next - The callback function to run after this function
+  *     finishes.
+ */
+const createUser = (data, next) => {
+  // Ensure the required data is available.
+  if (!data || !data.role || !data.email || !data.password ||
+      (data.role === 'Student' && !data.studentId)) {
+    next({ err: 'Required parameters not found.' });
+    return;
   }
 
-  // TODO(ryanfaulkenberry100): Write user to database.
+  switch (data.role) {
+    case "Student":
+      // Create a student
+      const student = new Student({
+        email: data.email,
+        password: data.password,
+        firstName: data.firstName,
+        lastName: data.lastName,
+        role: data.role,
+        studentId: data.studentId
+      });
 
-  return new response.ResponseObject(201, {"uid":"1", "url":"//www.google.com"});
-  // TODO(ryanfaulkenberry100): Return actual data.
-}
+      // Search for possible duplicate student.
+      Student.findOne({ email: student.email, studentId: student.studentId }, (err, existingUser) => {
+        if (err) {
+          next(err);
+          return;
+        }
+
+        if (existingUser) {
+          next({ err: 'Account with that email address or student id already exists.' });
+        } else {
+          student.save((err) => { next(err, student.studentId); });
+        }
+      });
+      break;
+    case "Admin":
+      // Create an admin
+      const admin = new Admin({
+        email: data.email,
+        password: data.password,
+        firstName: data.firstName,
+        lastName: data.lastName,
+        role: data.role
+      });
+
+      // Search for possible duplicate admin.
+      Admin.findOne({ email: admin.email }, (err, existingUser) => {
+        if (err) {
+          next(err);
+          return;
+        }
+
+        if (existingUser) {
+          next({ err: 'Account with that email address already exists.' });
+        } else {
+          // TODO(ryanfaulkenberry100): Does the admin need an ID?
+          admin.save((err) => { next(err); });
+        }
+      });
+      break;
+  }
```

Comment:
A lot of the admin and student code seems to be duplicated.  Is there a way to have some of this written once? You can probably just search for a duplicate user, rather than having separate statements for finding duplicate admins and students.